packages/react/src/components/builder-content.component.tsx

```diff
@@ -140,17 +140,17 @@ export class BuilderContent<ContentType extends object = any> extends React.Comp
         if (location.href.includes('builder.debug=true')) {
           eval('debugger');
         }
+        let newData = this.state.data as any;
         for (const patch of patches) {
-          applyPatchWithMinimalMutationChain(this.state.data, patch);
+          newData = applyPatchWithMinimalMutationChain(newData, patch, false);
         }
         this.setState({
           updates: this.state.updates + 1,
-          data: this.state.data ? { ...this.state.data } : this.state.data,
+          data: newData,
```

Comment:
It looks like the only change here besides putting `this.state.data` into a variable (`newData`) is that now we don't make data a new object with `{ ...this.state.data }`. Was that causing come of the performance issue?

Reply:
oh no this is unrelated, just was funky code. I first suspected something was wrong here, then realized it's actually ok, but I think this is the more "proper" functional approach